### PR TITLE
Lps 123057

### PR DESCRIPTION
**Pull request by mtambara**
Made some changes, let me know if this still fits your requirements

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:test:sfa

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
@jeyvison when you have a chance, please test these changes to see if they work for you. If you approve and CI passes, I can forward.

**Comment by drewbrokke** [comment]
@mtambara looks like a SF failure. I'll leave the pull open and you can just force-push an update.

**Comment by drewbrokke** [comment]
@mtambara nevermind, that was my fault - I made a typo while trying to trigger SF.

File: modules/apps/static/portal-file-install/portal-file-install-impl/src/main/java/com/liferay/portal/file/install/internal/properties/TypedProperties.java [hunk 1]
```diff
@@ -209,19 +208,6 @@ public boolean nextProperty() throws IOException {
 
 			value = InterpolationUtil.substVars(value.trim());
 
-			if (value.startsWith("LIFERAY_")) {
-				value = InterpolationUtil.substVars(
-					GetterUtil.getString(System.getenv(value)));
-
-				matcher = _propertyValuePattern.matcher(value);
```
Comment (drewbrokke) [inline modules/apps/static/portal-file-install/portal-file-install-impl/src/main/java/com/liferay/portal/file/install/internal/properties/TypedProperties.java]: @mtambara I think this is the only usage of the field, you'll need to remove it.

**Comment by drewbrokke** [comment]
ci:stop

**Comment by drewbrokke** [comment]
@mtambara one quick change requested, see my inline comment. I'll leave the pull open, so you can just force push an update.

**Comment by jeyvison** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mtambara** [comment]
Pushed the change

**Comment by mtambara** [comment]
ci:test:sf

**Comment by mtambara** [comment]
ci:test:relevant

**Comment by jeyvison** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by jeyvison** [comment]
Hi @mtambara , Thanks a lot for you changes, the code is a lot more cleaner now :)

Two notes:

- You changed back the line pattern validation so **LIFERAY_** variables are not a match anymore and therefore are ignored.
- The problem with removing the _propertyValuePattern validations is that if, for example, i define an Env variable like this: 
`LIFERAY_MY_ENV=!@#4XD` this property will probably not work and we will never know why since the retrieved value from **System.getEnv** is not validated anymore and there is no output for letting the developer know, as we have for line pattern matching.

Does it make sense?

**Comment by mtambara** [comment]
Hi @jeyvison:
For your first point, LIFERAY_ variables are not being ignored, as seen in ```testLoadEnvVariable```. They'll only be ignored if they're part of the key, but that can't be substituted so there isn't a reason to put the liferay prefix there.

For your second point, config values are allowed to be nearly any string as long as they are inside quotes, parentheses, or brackets. With my change, the substitution will still need to be inside ```${}``` for consistency, but that would already need to be inside quotes, parentheses or brackets and would be caught by the line pattern check if it wasn't.

For example, to substitute for an env var like ```LIFERAY_MY_ENV=!@#4XD```, you would need to have a config line like ```testKey = "${LIFERAY_MY_ENV}"``` which would get turned into ```testKey = "!@#4XD"```

**Comment by jeyvison** [comment]
@mtambara  Man, I completely misunderstood that! Now i got it! I just tested it here and it works perfectly :) Thanks a lot, man :)
@drewbrokke LGTM!

**Comment by drewbrokke** [comment]
ci:forward

**Comment by drewbrokke** [comment]
FYI @dejuknow

**Pull request closed without merging**